tests/test_url_handler.py

```diff
@@ -0,0 +1,696 @@
+    ]
+
+    assert url_handler.add_url(project, form_data) == expected
+
+
+def test_add_url_cannot_get_system_nominator():
+    form_data = {'url_value': 'http://www.example.com'}
+    project = factories.ProjectFactory()
+    models.Nominator.objects.get().delete()
+
+    with pytest.raises(http.Http404):
+        url_handler.add_url(project, form_data)
+
+
+@pytest.mark.xfail(reason='Called function surt_exists raises 404, does not return Falsy value.')
+def test_add_url_cannot_get_or_create_surt():
+    form_data = {'url_value': 'http://www.example.com'}
+    project = None
+    with pytest.raises(http.Http404):
+        url_handler.add_url(project, form_data)
+
+
+def test_add_url_cannot_get_or_create_nominator():
+    form_data = {
+        'url_value': 'http://www.example.com',
+        'nominator_email': None,
+        'nominator_name': None,
+        'nominator_institution': None
+    }
+    project = factories.ProjectFactory()
+
+    with pytest.raises(http.Http404):
+        url_handler.add_url(project, form_data)
+
+
+def test_add_metadata():
+    project = factories.ProjectWithMetadataFactory()
+    nominator = factories.NominatorFactory()
+    attribute_names = [x.name for x in project.metadata.all()]
+    value = 'some_value'
+    form_data = {
+        'nominator_email': nominator.nominator_email,
+        'scope': '1',
+        'url_value': 'http://www.example.com',
+        attribute_names[0]: value
+    }
+    expected = [
+        'You have successfully nominated {0}'.format(form_data['url_value']),
+        'You have successfully added the {0} "{1}" for {2}'.format(
+            attribute_names[0], value, form_data['url_value'])
+    ]
+
+    assert url_handler.add_metadata(project, form_data) == expected
+
+
+def test_add_metadata_nominator_does_not_exist_in_project():
+    project = factories.ProjectFactory()
+    form_data = {'nominator_email': 'someone@someplace.com'}
+
+    with pytest.raises(http.Http404):
+        url_handler.add_metadata(project, form_data)
+
+
+@pytest.mark.parametrize('url, expected', [
+    ('http://www.example.com', 'http://www.example.com'),
+    ('   http://www.example.com   ', 'http://www.example.com'),
+    ('https://www.example.com', 'http://www.example.com'),
+    ('http://http://www.example.com', 'http://www.example.com'),
+    ('http://www.example.com///', 'http://www.example.com')
+])
+def test_check_url(url, expected):
+    assert url_handler.check_url(url) == expected
+
+
+def test_get_nominator_when_nominator_exists():
+    nominator = factories.NominatorFactory()
+    form_data = {'nominator_email': nominator.nominator_email}
+    assert url_handler.get_nominator(form_data) == nominator
+
+
+def test_get_nominator_when_nominator_does_not_exist():
+    form_data = {
+        'nominator_email': 'somebody@somewhere.com',
+        'nominator_name': 'John Smith',
+        'nominator_institution': 'UNT'
+    }
+
+    assert len(models.Nominator.objects.all()) == 1
```

Comment:
Ah, yes, there always exists a system nominator. It is created by the initial data fixture.